R/declare_diagnosands.R

```diff
@@ -0,0 +1,225 @@
+#' @param ... A set of new diagnosands
+#' @param select A set of the default diagnosands to report e.g., \code{select = c(bias, rmse)}
+#' @param subtract A set of the default diagnosands to NOT report e.g., \code{subtract = c(bias, rmse)}
+#' @param keep_defaults A flag for whether to report the default diagnosands. Defaults to \code{TRUE}.
+#' @param subset A subset of the simulations data frame within which to calculate diagnosands e.g. \code{subset = p < .05}.
+#' @param alpha Alpha significance level. Defaults to \code{.05}.
+#' @param label Label for the set of diagnosands.
+#' @param data a data.frame
+#'
+#' @details
+#'
+#' If coefficients is TRUE, the names of ... will be returned in a `coefficients` column, and `estimand_label`
+#' will contain the step label. This can be used as an additional dimension for use in diagnosis.
+#'
+#'
+#' @importFrom rlang eval_tidy quos  is_quosure quo_is_call
+#' @rdname declare_diagnosands
+diagnosand_handler <- function(data, ..., 
+                               select,
+                               subtract,
+                               keep_defaults = TRUE,
+                               subset = NULL, 
+                               alpha = 0.05,
+                               label) {
+  
+  options <- quos(...)
+  if (length(options) > 0 && names(options)[1] == "") names(options)[1] <- label
+
+  # subsetting the data -----------------------------------------------------
+
+  subset <- substitute(subset)
+  if (is_quosure(subset)) subset <- subset[[2]]
+  idx <- eval_tidy(subset, data = data)
+  if (!is.null(idx)) {
+    data <- data[idx, , drop = FALSE]
+  }
+
+  # defaults ----------------------------------------------------------------
+  
+  defaults_quos <- 
+    quos(
+      bias = mean(est - estimand),
+      rmse = sqrt(mean((est - estimand) ^ 2)),
+      power = mean(p < alpha),
+      coverage = mean(estimand <= ci_upper & estimand >= ci_lower),
+      mean_estimate = mean(est),
+      sd_estimate = sd(est),
+      mean_se = mean(se),
+      type_s_rate = mean((sign(est) != sign(estimand))[ p < alpha ] ),
+      mean_estimand = mean(estimand)
+    )
+  
+  if (!missing(select)) {
+    select_quo <- enquo(select)
+    if (quo_is_call(select_quo)) {
+      select_set <- as.character(call_args(select_quo))
```

Comment:
use the existing helper functions for converting to character.

Reply:
ooh, thanks, which are those?